### PR TITLE
Simple web crud

### PR DESCRIPTION
**Pull request by cosmiclavaflow**

### CODE REVIEW

File: src/main/java/com/music/tagger/controller/RestResponseEntityExceptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@Slf4j
+@ControllerAdvice
+public class RestResponseEntityExceptionHandler extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler({TrackNotFoundException.class, IllegalStateException.class})
+    public ResponseEntity<Object> handleInternal(Exception ex, WebRequest request) {
+        log.warn("406 Status Code", ex);
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/controller/RestResponseEntityExceptionHandler.java:19]: 406 is not   HttpStatus.INTERNAL_SERVER_ERROR(500)!
Why TrackNotFoundException.class match to 500?

File: src/main/java/com/music/tagger/controller/RootTrackController.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import static com.music.tagger.utils.ComplexDomainObjectCreator.getEmptyTrack;
+
+@RestController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class RootTrackController {
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/controller/RootTrackController.java:26]: Please add a root path to the controller.

File: src/main/java/com/music/tagger/exceptions/TrackNotFoundException.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.music.tagger.exceptions;
+
+public class TrackNotFoundException extends Exception {
+
+
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/exceptions/TrackNotFoundException.java]: remove the redundant blank line.

File: src/main/java/com/music/tagger/exceptions/TrackNotFoundException.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    public TrackNotFoundException() {
+        super();
+    }
+
+
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/exceptions/TrackNotFoundException.java]: remove the redundant blank line.

File: src/main/java/com/music/tagger/persistence/entity/Album.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@Entity
+@Table(name = "album")
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/persistence/entity/Album.java]: in general, the name of the table is in the plural.

File: src/main/java/com/music/tagger/persistence/entity/Artist.java [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.ArrayList;
+import java.util.List;
+
+@Data
+@Entity
+@Table(name = "artist")
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/persistence/entity/Artist.java]: the same

File: src/main/java/com/music/tagger/persistence/entity/Track.java [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.Table;
+
+@Data
+@Entity
+@Table(name = "track")
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/persistence/entity/Track.java]: the same

**Review by Anderwerty — CHANGES_REQUESTED** [review]
Please resolve comments and add tests.

File: src/main/java/com/music/tagger/controller/RestResponseEntityExceptionHandler.java (hunk 1 above)
Comment (cosmiclavaflow) [reply src/main/java/com/music/tagger/controller/RestResponseEntityExceptionHandler.java:19]: This is from a copied block of code. I forgot to change it when I chose the response code that better fits this method.

File: src/main/java/com/music/tagger/config/RootMvcConfig.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    public ModelMapper modelMapper() {
+        ModelMapper mm = new ModelMapper();
+
+        PropertyMap<SimpleTrackDto, Track> propertyMap = new PropertyMap<SimpleTrackDto, Track> (){
+            protected void configure() {
+                map().setName(source.getTrackName());
+                map().getAlbum().setName(source.getAlbumName());
+                map().getArtist().setName(source.getArtistName());
+
+                //force to skip mapper relation mapping fields
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/config/RootMvcConfig.java]: Do we need this comment?

File: src/main/java/com/music/tagger/controller/RootTrackController.java [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+
+import static com.music.tagger.utils.ComplexDomainObjectCreator.getEmptyTrack;
+
+@Controller
+@RequestMapping("/")
```
Comment (Anderwerty) [inline src/main/java/com/music/tagger/controller/RootTrackController.java:24]: / -> trank or music

**Review by Anderwerty — CHANGES_REQUESTED** [review]

**Review by Anderwerty — APPROVED** [review]

**Pull request merged**